File: Dipti/DP-1.js
// // calculate the sum and average 
function calculateSumAndAverage(dataPoints) {
    // Split the input string into an array of numbers
    const numbers = dataPoints.split(' ').map(Number);
  
    // Calculate the sum of the numbers
    const sum = numbers.reduce((acc, num) => acc + num, 0);
  
    // Calculate the average of the numbers
    const average = sum / numbers.length;
  
    // Return an object containing the sum and average
    return {
      sum,
      average
    };
  }
  
  // Function to prompt the user for input and display the results
  const prompt = require("prompt-sync")({sigint: true});
  function processDataPoints() {
    const dataPoints = prompt('Please enter a series of numerical data points separated by spaces:');
    
    // Check if input is provided
    if (dataPoints.trim() === '') {
      console.log('No data points provided.');
      return;
    }
    
    // Calculate the sum and average
    const result = calculateSumAndAverage(dataPoints);
    
    // Display the results
    console.log('Thank you for providing the data points.');
    console.log('The sum of the numbers is:', result.sum);
    console.log('The average of the numbers is:', result.average.toFixed(2));
  }
  
  // Call the function to start the program
  processDataPoints();

File: Niharika/dp1.py
def calculate_sum(numbers):
    """
    Calculates the sum of a series of numbers.

    Args:
        numbers (list): A list of numbers.

    Returns:
        float: The sum of the numbers.
    """
    return sum(numbers)
  

def calculate_average(numbers):
    """
    Calculates the average of a series of numbers.

    Args:
        numbers (list): A list of numbers.

    Returns:
        float: The average of the numbers.
    """
    return sum(numbers) / len(numbers)
def main():
    """
    Main function to run the data analytics program.
    """
    print("Welcome to QuantFinTech Data Analytics!")
    data_points = input("Please enter a series of numerical data points separated by spaces:\n")

    # Convert the input string into a list of numbers
    numbers = [float(num) for num in data_points.split()]

    # Calculate the sum and average
    total_sum = calculate_sum(numbers)
    average = calculate_average(numbers)

    print("\nThank you for providing the data points.")
    print("The sum of the numbers is:", total_sum)
    print("The average of the numbers is:", average)


# Run the program
if __name__ == '__main__':
    main()